tools/jbrowse2/jbrowse2.py

```diff
@@ -0,0 +1,1619 @@
+        )
+        metadata["history_user_email"] = '<a href="mailto:{0}">{0}</a>'.format(
+            metadata["history_user_email"]
+        )
+        metadata["hist_name"] = metadata["history_display_name"]
+        metadata[
+            "history_display_name"
+        ] = '<a target="_blank" href="{galaxy}/history/view/{encoded_hist_id}">{hist_name}</a>'.format(
+            galaxy=GALAXY_INFRASTRUCTURE_URL,
+            encoded_hist_id=metadata.get("history_id", "not available"),
+            hist_name=metadata.get("history_display_name", "not available"),
+        )
+    if node.findall("tool"):
+        for (key, value) in node.findall("tool")[0].attrib.items():
+            metadata["tool_%s" % key] = value
+        metadata[
+            "tool_tool"
+        ] = '<a target="_blank" href="{galaxy}/datasets/{encoded_id}/show_params">{tool_id}{tool_version}</a>'.format(
+            galaxy=GALAXY_INFRASTRUCTURE_URL,
+            encoded_id=metadata.get("dataset_id", ""),
+            tool_id=metadata.get("tool_tool_id", ""),
+            tool_version=metadata.get("tool_tool_version", ""),
+        )
+    return metadata
+
+
+class JbrowseConnector(object):
+    def __init__(self, outdir, jbrowse2path):
+        self.assemblies = []  # these require more than a few line diff.
+        self.assmeta = {}
+        self.giURL = GALAXY_INFRASTRUCTURE_URL
+        self.outdir = outdir
+        self.genome_firstcontig = None
+        self.jbrowse2path = jbrowse2path
+        os.makedirs(self.outdir, exist_ok=True)
+        self.genome_names = []
+        self.trackIdlist = []
+        self.tracksToAdd = {}
+        self.config_json = {}
+        self.config_json_file = os.path.join(outdir, "config.json")
+        self.clone_jbrowse()
+
+    def get_cwd(self, cwd):
+        if cwd:
+            return self.outdir
+        else:
+            return subprocess.check_output(["pwd"]).decode("utf-8").strip()
+            # return None
+
+    def subprocess_check_call(self, command, output=None, cwd=True):
+        if output:
+            log.debug("cd %s && %s >  %s", self.get_cwd(cwd), " ".join(command), output)
+            subprocess.check_call(command, cwd=self.get_cwd(cwd), stdout=output)
+        else:
+            log.debug("cd %s && %s", self.get_cwd(cwd), " ".join(command))
+            subprocess.check_call(command, cwd=self.get_cwd(cwd))
+
+    def subprocess_popen(self, command, cwd=True):
+        log.debug(command)
+        p = subprocess.Popen(
+            command,
+            cwd=self.get_cwd(cwd),
+            shell=True,
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        output, err = p.communicate()
+        retcode = p.returncode
+        if retcode != 0:
+            log.error(command)
+            log.error(output)
+            log.error(err)
+            raise RuntimeError("Command failed with exit code %s" % (retcode))
```

Comment:
```suggestion
            raise RuntimeError(f"Command ( {command} ) failed with exit code {retcode}")
```

Reply:
Yes - same problem - leftover. `Not my fault@!!@#`